libs/util/nestjs/tracing/src/lib/tracing.module.ts

```diff
@@ -0,0 +1,155 @@
+      idGenerator: new AWSXRayIdGenerator(),
+    };
+  }
+
+  private static addLocalConfiguration(
+    configuration: Partial<OpenTelemetryModuleConfig>
+  ): Partial<OpenTelemetryModuleConfig> {
+    const collectorOptions = {
+      // url is optional and can be omitted - default is http://localhost:4318/v1/traces,
+      // url: '<opentelemetry-collector-url>',
+    };
+    const traceExporter = new OTLPTraceExporter(collectorOptions);
+
+    return {
+      ...configuration,
+      spanProcessor: new BatchSpanProcessor(traceExporter),
+      traceExporter,
+    };
+  }
+
+  private static createDefaultConfiguration(
+    configuration: Partial<OpenTelemetryModuleConfig>
+  ): Partial<OpenTelemetryModuleConfig> {
+    const autoInstrumentations = getNodeAutoInstrumentations({
+      "@opentelemetry/instrumentation-fs": {
+        requireParentSpan: true,
+        enabled: true,
+        createHook: (funtionName, { args }) => {
+          return !args[0].toString().indexOf("node_modules");
+        },
+        endHook: (funtionName, { args, span }) => {
+          span.setAttribute("file", args[0].toString());
+        },
+      },
+      "@opentelemetry/instrumentation-http": {
+        requireParentforOutgoingSpans: true,
+        requestHook: (span: Span, request: ClientRequest) => {
+          span.updateName(`${request.method} ${request.path}`);
+        },
+        enabled: true,
+        ignoreIncomingPaths: ["/health", "/_health", "/healthz", "healthcheck"],
+      },
+      "@opentelemetry/instrumentation-net": {
+        enabled: false,
+      },
+      "@opentelemetry/instrumentation-dns": {
+        enabled: false,
+      },
+      "@opentelemetry/instrumentation-graphql": {
+        enabled: true,
+        mergeItems: true,
+        ignoreTrivialResolveSpans: true,
+        depth: 2,
+      },
+      "@opentelemetry/instrumentation-express": {
+        enabled: true,
+        ignoreLayersType: [ExpressLayerType.MIDDLEWARE],
+      },
+    });
+
+    configuration = {
+      ...configuration,
+      instrumentations: [
+        autoInstrumentations,
+        new KafkaJsInstrumentation({}),
+        new PrismaInstrumentation({}),
+      ],
+      traceAutoInjectors:
+        configuration?.traceAutoInjectors ||
+        OpenTelemetryModuleDefaultConfig.traceAutoInjectors,
+    };
+
+    return configuration;
+  }
+
+  static forRoot(
+    configuration?: Partial<OpenTelemetryModuleConfig>
+  ): Promise<DynamicModule> {
+    const commonConfiguration = this.createDefaultConfiguration(
+      configuration ?? {}
+    );
+    return OpenTelemetryModule.forRoot(
+      process.env.NODE_ENV === "production"
+        ? this.addAwsXRayConfiguration(commonConfiguration)
+        : this.addLocalConfiguration(commonConfiguration)
+    );
+  }
+
+  static async forRootAsync(
+    configuration?: OpenTelemetryModuleAsyncOption
+  ): Promise<DynamicModule> {
+    return OpenTelemetryModule.forRootAsync({
+      ...configuration,
```

Comment:
This might still be a problem with `...configuration` if `configuration` is `undefined`

Reply:
fixed